### PR TITLE
add reader example to C++ protobuf guide

### LINKED ISSUE #460: Guide: Reading Protobuf data to/from MCAP files in C++

### PR DESCRIPTION
**Pull request by james-rms**
**Public-Facing Changes**

**Description**

Fixes #460

### CODE REVIEW

**Comment by esthersweon** [comment]
The length of this page makes me wonder if we should break out reading and writing onto separate pages.

i.e. "Reading Protobuf" and "Writing Protobuf"

File: docs/guides/cpp/protobuf.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+```cpp
+foxglove::PosesInFrame path_msg;
+// Fill in path_msg.
+std::string data = path_msg.SerializeAsString();
+
+mcap::Message msg;
+msg.channelId = path_channel.id;
+msg.logTime = timestamp_ns;
+msg.publishTime = msg.logTime;
+msg.data = reinterpret_cast<const std::byte*>(data.data());
```
[the remark this answers is not in the document]
Comment (esthersweon) [reply docs/guides/cpp/protobuf.md:83]: Related to https://github.com/foxglove/website/pull/665

[the remark this answers is not in the document]
Comment (james-rms) [reply docs/guides/cpp/protobuf.md:107]: @esthersweon i've filled in the example code i sent you previously, i think it's better to have something there.

**Comment by james-rms** [comment]
I think the context from the "writer" is useful to have when looking at the "reader" part of the guide, so I'll leave them together for now.

**Pull request merged**